src/App.tsx

```diff
@@ -123,8 +129,10 @@ function App() {
     };
 
     const handleSortChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
-        const [sort, order] = event.target.value.split('_') as [string, string];
-        setSortAndOrder(sort as any, order as any);
+        const [sort, order] = event.target.value.split('_');
+        if (validSortFields.includes(sort as SortField) && validSortOrders.includes(order as SortOrder)) {
+            setSortAndOrder(sort as SortField, order as SortOrder);
```

Comment:
Instead of type-casting ("as" operator) in such cases it is better and more convenient to use type0guards ("is" operator). You may check this out